src/werkzeug/datastructures.py

```diff
@@ -3058,7 +3058,9 @@ def save(self, dst, buffer_size=16384):
         from shutil import copyfileobj
 
         close_dst = False
-        dst = fspath(dst)
+
+        if hasattr(dst, "__fspath__"):
```

Comment:
This breaks support for `pathlib` in Python 3.6, which doesn't have the `__fspath__` API yet.

Reply:
@davidism  both `PathLike` and `fspath` were introduced in 3.6 (per PEP 519), and `PathLike` has `__fspath__`, or I'm missing something?

Reply:
You're right, I meant < 3.6. 1.0.x supports 3.5 still.

Reply:
There shouldn't be problems with 3.5 too - if there is no `__fspath__` attribute, then `fspath` is not touched and it goes to the original logic: check string types etc..

Reply:
But there are `Path` objects in 3.5, they just don't have `__fspath__`. That's why we have the `_compat.fspath` function.

Reply:
I'm fine with just saying "pathlib is only supported on 3.6+", and the whole thing goes away once we drop Python < 3.6. Just want to be clear on what's intended.

Reply:
Ah, you mean `pathlib` objects.. But I don't see how current `_compat.fspath` implementation supports `pathlib` objects prior 3.6. IMO to make them really work for < 3.6 it's needed to add something like `if isinstance(dst, pathlib.PurePath): dst = dst.open('wb')` in `save` method, right?

Reply:
I had assumed our compat was doing `str(path)` on things that didn't have `__fspath__`, but it's not. So I guess I already had this conversation and came to the same conclusion of not supporting 3.5's pathlib in the first place.